### PR TITLE
Docker setup for swiftserver CI

### LINKED ISSUE #16: Test against all supported Linux platforms in CI
#9 introduces a barebones Actions configuration that only tests on Ubuntu 20.04. We should update this to cover all of the Linux platforms mentioned on https://download.swift.org.

### PR DESCRIPTION
**Pull request by patrickfreed**
Closes #16 

This PR introduces the necessary docker-compose files for using https://ci.swiftserver.group/ for CI.

### CODE REVIEW

File: Tests/SwiftlyTests/InstallTests.swift [hunk 1]
```diff
@@ -35,7 +35,7 @@ final class InstallTests: SwiftlyTests {
     /// Tests that `swiftly install a.b` installs the latest patch version of Swift a.b.
```
Comment (patrickfreed) [inline Tests/SwiftlyTests/InstallTests.swift:35]: Swift 5.6 isn't available on Ubuntu 22.04, so I updated these tests to use 5.7.x wherever 5.6 or older was used.

**Comment by yim-lee** [comment]
@swift-server-bot add to allowlist

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Comment by yim-lee** [comment]
@swift-server-bot add to allowlist

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Comment by yim-lee** [comment]
@swift-server-bot test install please

**Comment by yim-lee** [comment]
@swift-server-bot test install please

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Comment by yim-lee** [comment]
@swift-server-bot test install please

**Review by yim-lee — COMMENTED** [review]
Sorry it was my mistake. I don't think we need this change.

File: docker/docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    depends_on: [test-setup]
+    volumes:
+      - ~/.ssh:/root/.ssh
+      - ..:/code:z
+    working_dir: /code
+    cap_drop:
+      - CAP_NET_RAW
+      - CAP_NET_BIND_SERVICE
+    environment:
+      - SWIFTLY_GITHUB_TOKEN="$GITHUB_ACCESS_TOKEN"
```
Comment (yim-lee) [inline docker/docker-compose.yaml:25]: Sorry it was my mistake. I don't think we need this change.

**Comment by patrickfreed** [comment]
@swift-server-bot test install please

**Comment by patrickfreed** [comment]
@swift-server-bot test install please

**Comment by patrickfreed** [comment]
@yim-lee A few more questions/requests:
- Could we add a separate "lint" task? This could be run once through `docker/docker-compose.yaml` alone and not need to be done on all the platforms.
- I've just added files for RHEL9, can we add test/install test tasks for that platform?
- If I wanted to upgrade the Swift version that all the existing platforms used, would simply updating the docker-compose filenames make that happen automatically? Or would it be a more manual process?

Thanks again for all your help!

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Comment by yim-lee** [comment]
@swift-server-bot test install please

**Comment by yim-lee** [comment]
@patrickfreed 

> Could we add a separate "lint" task? This could be run once through docker/docker-compose.yaml alone and not need to be done on all the platforms.

Done. The job is called "soundness".

> I've just added files for RHEL9, can we add test/install test tasks for that platform?

Done.

> If I wanted to upgrade the Swift version that all the existing platforms used, would simply updating the docker-compose filenames make that happen automatically? Or would it be a more manual process?

With the current setup it would require manual changes. We could potentially rename the docker files and CI jobs and remove the Swift version from them, if you don't foresee having more than one job per platform? (It's also fine if we have to update CI with every Swift upgrade, that's what we do for other projects.)

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Comment by yim-lee** [comment]
@swift-server-bot test install please

**Comment by yim-lee** [comment]
@swift-server-bot test this please

**Comment by yim-lee** [comment]
@patrickfreed I think we are all good now 🚀

**Comment by patrickfreed** [comment]
awesome, thanks again for all your help!

File: docker/test-amazonlinux2.dockerfile [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+COPY ./scripts/install-libarchive.sh /
+RUN /install-libarchive.sh
+
+# tools
+RUN mkdir -p $HOME/.tools
+RUN echo 'export PATH="$HOME/.tools:$PATH"' >> $HOME/.profile
+
+# swiftformat (until part of the toolchain)
+
+ARG swiftformat_version=0.49.18
```
Comment (adam-fowler) [inline docker/test-amazonlinux2.dockerfile:22]: Why is SwiftFormat installed in the amazonlinux and ubi9 Dockerfiles but not the standard test Dockerfile

**Review by adam-fowler — APPROVED** [review]
Only one question but otherwise looks good

File: docker/test-amazonlinux2.dockerfile (hunk 3 above)
Comment (patrickfreed) [reply docker/test-amazonlinux2.dockerfile]: Oh good catch, that was just an oversight. We have the separate lint/soundness dockerfile, so the test dockerfiles don't need to install swiftformat anymore. Removed.

**Pull request merged**